pynamodb/attributes.py

```diff
@@ -212,14 +219,14 @@ def _initialize_attributes(cls):
 @add_metaclass(AttributeContainerMeta)
 class AttributeContainer(object):
 
-    def __init__(self, **attributes):
+    def __init__(self, _previously_saved = False, **attributes):
```

Comment:
@garrettheel can you think of any more elegant way of letting the model know whether it's a newly constructed one or a deserialized one?

The tricky part is that it needs to be known during initialization, since that's where defaults are established, and we thought of introducing a concept of `default_for_new` i.e. defaults that only apply during initialization (of ostensibly new models) but don't get applied to existing models that are going through a load-save cycle:
```proto
class MyModel(Model):
    time_to_live = TTLAttribute(default_for_new=timedelta(days=366))
```

Unlike what was previously claimed, this feature cannot be re-implemented as a custom initializer -- see why:
https://gist.github.com/ikonst/ebc4b7f54079afedd7aba452544c1c71
Instead, you'd have to implement it in application code (rather than in the model):
```python
def my_app_handler(foo, bar):
    my_model = MyModel()
    my_model.id = str(uuid4())
    my_model.foo = foo
    my_model.bar = bar
    my_model.time_to_live = timedelta(days=366)
    my_model.save()
```

Reply:
In your gist.. Get calls the constructor of the model (so the value of the attribute becomes 42). https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/models.py#L404 calls https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/models.py#L438.

Refresh does not call the constructor - it just gets the values from dynamo and overwrites the model's attributes https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/models.py#L380.  Hence why the value is 7.

Reply:
Yeah, my case isn't even a good simulation of the actual scenario we're concerned about. Here's something more resembling our concern:
https://gist.github.com/ikonst/66de10eb86721ac343da581aed24ff82

Reply:
I think your gists basically validate the concern that putting something in the constructor of the model will not work given the current pynamo codebase